crates/sovereign/sovereign-integration-tests/src/impls/build_rollup_driver.rs

```diff
@@ -40,20 +43,24 @@ where
     ) -> Result<SovereignRollupDriver, Bootstrap::Error> {
         let rpc_config = &node_config.runner.rpc_config;
 
-        let rpc_client = HttpClientBuilder::default()
-            .build(format!(
-                "http://{}:{}",
-                rpc_config.bind_host, rpc_config.bind_port
-            ))
-            .map_err(Bootstrap::raise_error)?;
-
-        let subscription_client = WsClientBuilder::default()
-            .build(format!(
-                "ws://{}:{}",
-                rpc_config.bind_host, rpc_config.bind_port
+        let (rpc_client, subscription_client) = retry(ExponentialBackoff::default(), || async {
+            Ok((
+                HttpClientBuilder::default()
+                    .build(format!(
+                        "http://{}:{}",
+                        rpc_config.bind_host, rpc_config.bind_port
+                    ))
+                    .map_err(Bootstrap::raise_error)?,
+                WsClientBuilder::default()
+                    .build(format!(
+                        "ws://{}:{}",
+                        rpc_config.bind_host, rpc_config.bind_port
+                    ))
+                    .await
+                    .map_err(Bootstrap::raise_error)?,
             ))
-            .await
-            .map_err(Bootstrap::raise_error)?;
+        })
```

Comment:
should we split `HttpClient` and `WsClient` in two `retry`s?

Reply:
I think a simple while loop with short timeout should be sufficient. We don't need complicated retry logic here, because the node should still start reasonably fast. In fact, having longer or infinite retry will cause failure to be very hard to catch or detected. If the node do not start up within the few seconds that we expect, then the test should still fail so that we can be notified. The retry logic should be there more so that when the node _do_ start up faster, then the test can resume slightly quicker.

Reply:
It should be sufficient to retry only on the construction of the first client, i.e. HTTP client. Once the node has started, the construction of the WS client followed should just succeed.

Reply:
For me, the error actually happens at wsclient.

Nonetheless, I can remove the `retry` dependency with bounded timeout.